mantidproject/mantid
# Remove extra none in the legend of 1D plots

**Description of work.**
If the legend of a 1D plot was modified (by re-plotting data with errors for example) on Ubuntu an extra 'none' entry appeared in the legend. This was due to older versions of Matplotlib casting None to a string, see the commit https://github.com/matplotlib/matplotlib/commit/84292c46902913273d9231414e19bd11664117a6

This PR adds in a check before the legend is generated, to remove the 'None' string.



**To test:**

1. Open workbench on Ubuntu ( but presumably other distributions (and possibly mac?)
2. Plot any data
3. Right-click and add errors
4. A spurious 'None' entry should no longer appear. 



Fixes #28090 . 




---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
